feg/gateway/services/s8_proxy/servicers/s8_proxy.go

```diff
@@ -220,3 +241,46 @@ func validateDeleteSessionRequest(dsr *protos.DeleteSessionRequestPgw) error {
 	}
 	return nil
 }
+
+// GetHealthStatus retrieves a health status object which contains the current
+// health of the service
+func (s *S8Proxy) GetHealthStatus(ctx context.Context, req *orc8r_protos.Void) (*protos.HealthStatus, error) {
```

Comment:
I'm a bit confused about how the health metrics are supposed to work in general (also the existing metrics for s6a). We have a public method in the service but it's not part of the interface and there are no callers?

Reply:
I am also unsure how this exactly happens, however, it appears that the proxies (other than the one I wrote myself...) implement the `ServiceHealthServer` interface in `feg/cloud/go/protos/service_health.pb.go`, which is used in some other parts.

I assume there is some magic happening that makes it all work in the end.

Reply:
disable/enable functions were created for HA. When health was bad. orc8r will send a disable message to the servicer, and enable the other HA. That only applies to Diameter services because they have some state. For GTP services all the state is at MME, so disable, enable does not matter.

The 3 functions here define the health interface. That is needed so the service can be added as a `RegisterServiceHealthServer`